### PR TITLE
Fix sf3

### PR DESCRIPTION
**Pull request by blump**

### CODE REVIEW

File: Form/Type/FileType.php [hunk 1]
```diff
@@ -56,7 +56,8 @@ public function __construct($class, UploadFileHelperInterface $uploadFileHelper)
      */
     public function getParent()
     {
-        return 'file';
+        $type = method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix') ? 'Symfony\Component\Form\Extension\Core\Type\FileType' : 'file';
```
Comment (dantleech) [inline Form/Type/FileType.php]: I wonder if there is a smoother way to determine the SF version, /cc @WouterJ @dbu

**Comment by blump** [comment]
I'm sorry for the number of commit. :(

File: Tests/Resources/app/config/config.php [hunk 2]
```diff
@@ -12,5 +12,9 @@
 $container->setParameter('cmf_testing.bundle_fqn', 'Symfony\Cmf\Bundle\MediaBundle');
 $loader->import(CMF_TEST_CONFIG_DIR.'/default.php');
 $loader->import(CMF_TEST_CONFIG_DIR.'/phpcr_odm.php');
-$loader->import(__DIR__.'/cmf_media.yml');
+if (version_compare(strtolower(\Symfony\Component\HttpKernel\Kernel::VERSION), '3.0', '<')) {
+    $loader->import(__DIR__.'/cmf_media.yml');
+} else {
+    $loader->import(__DIR__.'/cmf_media_3_0.yml');
```
Comment (dbu) [inline Tests/Resources/app/config/config.php]: would it maybe be possible to extract the things that are specific to symfony 2 resp 3 into a separate file each that then imports the base file with the rest of configuration? something like cmf_media_2.yml:
```yaml
imports:
    - { resource: cmf_media.yml }

twig:
    form:
        resources:
            - 'CmfMediaBundle:Form:fields.html.twig'
```

File: composer.json [hunk 3]
```diff
@@ -13,15 +13,15 @@
...
+        "symfony/framework-bundle": "~2.3|~3.0"
     },
     "require-dev": {
         "twig/twig": "^1.12|^2.0",
         "symfony-cmf/testing": "^1.3",
         "doctrine/phpcr-odm": "^1.3",
         "doctrine/annotations": "^1.1.2",
         "jms/serializer-bundle": "^0.12|^1.0",
-        "liip/imagine-bundle": "~1.0.4",
+        "liip/imagine-bundle": "~1.5.3",
```
Comment (dbu) [inline composer.json]: are we no longer compatible with imagine-bundle 1.0.*? if its to allow a version that supports symfony 3, lets change the line to `^1.0.4`

**Comment by dbu** [comment]
thanks for the contribution! don't worry about the commits, we can squash them into one while merging, so all fine. i added a few things i would like to clean up, but apart from that looks ready to me.

**Comment by WouterJ** [comment]
Looks great!

However, we've abandoned this bundle as we have a too small team to maintain many bundles. So we either need to find a new maintainer of this bundle or I'm not sure if we can merge this feature (not because this is a too big feature to maintain, but because we have to maintain the complete bundle).

**Comment by dbu** [comment]
ah indeed, we abandoned the bundle. but i don't see much harm in continuing to merge such things as long as we find time to review them. its not like we deprecated this bundle in favor of something else...

@blump if you use this bundle in projects, how would you feel about taking over maintenance for it? it would mean you review pull requests if any come, and try to reply to issues that are opened. and possibly also work on improving the bundle, but that is no requirement.

File: composer.json [hunk 4]
```diff
@@ -13,15 +13,15 @@
...
+        "symfony/framework-bundle": "~2.3|~3.0"
     },
     "require-dev": {
         "twig/twig": "^1.12|^2.0",
         "symfony-cmf/testing": "^1.3",
         "doctrine/phpcr-odm": "^1.3",
         "doctrine/annotations": "^1.1.2",
         "jms/serializer-bundle": "^0.12|^1.0",
-        "liip/imagine-bundle": "~1.0.4",
+        "liip/imagine-bundle": "~1.0,>=1.0.4",
```
Comment (dbu) [inline composer.json]: this should be `^1.0.4`. that is the short form of what you are saying here.

**Comment by dbu** [comment]
can you fix the imagine-bundle dependency? then i will merge and tag as 1.3.1 as there is no other changes to it and i see no harm in doing this even if we don't actively maintain the bundle anymore.

**Comment by blump** [comment]
@dbu I hesitated but I do not use this time in my projects ... I had some time yesterday and someone asked for a fix to use it on SF3 so I told myself I was going do it.

I'm sorry, but I have enough time to be able to seriously work on this bundle ...

**Comment by dbu** [comment]
tagged 1.3.1, thanks @blump 

ok, no worries, just thought i'd ask... i much prefer a honest answer over an illusion of having somebody who maintains this bundle but then doesn't have the time to do it.

**Pull request merged**